src/Framework/Sdk/ReferencedSdk.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Linq;
+
+namespace Microsoft.Build.Framework
+{
+    /// <summary>
+    ///     Represents a software development kit (SDK) that is referenced in a &lt;Project /&gt; or &lt;Import /&gt; element.
+    /// </summary>
+    public sealed class ReferencedSdk : IEquatable<ReferencedSdk>
+    {
+        /// <summary>
+        ///     Initializes a new instance of the ReferencedSdk class.
+        /// </summary>
+        /// <param name="name">The name of the SDK.</param>
+        /// <param name="version">The version of the SDK.</param>
+        public ReferencedSdk(string name, string version)
+        {
+            Name = name;
+            Version = version;
+        }
+
+        /// <summary>
+        ///     Gets the name of the SDK.
+        /// </summary>
+        public string Name { get; }
+
+        /// <summary>
+        ///     Gets the version of the SDK.
+        /// </summary>
+        public string Version { get; }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="other"></param>
+        /// <returns></returns>
+        public bool Equals(ReferencedSdk other)
+        {
+            if (ReferenceEquals(null, other))
+                return false;
+            if (ReferenceEquals(this, other))
+                return true;
+            return string.Equals(Name, other.Name) && Equals(Version, other.Version);
+        }
+
+        /// <summary>
+        ///     Attempts to parse the specified string as a <see cref="ReferencedSdk" />.  The expected format is "SDK/Version".
+        ///     Values are not required to specify a version.
+        /// </summary>
+        /// <param name="sdk">An SDK name and version to parse in the format "SDK/Version" or just "SDK".</param>
+        /// <param name="referencedSdk">A parsed <see cref="ReferencedSdk" /> if the specified value is a valid SDK name.</param>
+        /// <returns><code>true</code> if the SDK name was successfully parsed, otherwise <code>false</code>.</returns>
+        public static bool TryParse(string sdk, out ReferencedSdk referencedSdk)
+        {
+            referencedSdk = null;
+
+            if (!string.IsNullOrWhiteSpace(sdk))
+            {
+                var parts = sdk.Split('/').Select(i => i.Trim()).ToArray();
```

Comment:
What about the range syntax (e.g.`FooSDK/min:3.4`)? I guess MSBuild should be doing that, otherwise all resolvers would duplicate the logic.